### PR TITLE
ami2 script for lcls-1 hutches

### PR DESCRIPTION
**Pull request by vespos**
## Description
Make a script to restart/stop ami2 and start more clients

## Motivation and Context
Needed for smoother operation with ami2 in LCLS-I hutches

## How Has This Been Tested?
With XPP DAQ and AMI

## Where Has This Been Documented?
N/A

### CODE REVIEW

File: scripts/startami2 [hunk 1]
```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+# ----------- Functions -----------
+get_status(){
+  grep ami2_ $STATUS_FILE | awk {'print $1"\t"$2"\t"$3'}
+}
+echo
```
Comment (ZLLentz) [inline scripts/startami2:7]: Is this left here by accident, or does it do something?

File: scripts/startami2 [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+fi
+
+if [[ $# -lt 2 ]]; then
+  HUTCH=${HOSTNAME:0:3}
+  HUTCHES=("xpp" "xcs" "mfx" "cxi" "mec")
+  if [[ ${HUTCHES[@]} =~ $HUTCH ]]; then
+    echo "Valid hutch found: $HUTCH"
+  else
+    echo "Invalid hutch: $HUTCH"
+    exit 1
```
Comment (ZLLentz) [inline scripts/startami2:72]: Tiny suggestion: we can tell the user which hutches are allowed or why their hutch is not valid

**Review by ZLLentz — COMMENTED** [review]
I took a look because I was curious and it seems pretty clean to me with respect to code health.

The shellcheck CI has a bunch of minor suggestions, mostly based around quoting variables and some related things.

**Comment by vespos** [comment]
Thanks for your comments Zach! Yes I noticed I did not have pre-commit installed on my checkout and was going to fix the syntax suggestions.
I will also address the few points you are raising here.

**Comment by vespos** [comment]
Still need to figure out some issues with the mypy deamon for the second client to work smoothly... Let's hold on merging until I figure this out

**Comment by vespos** [comment]
Ok, this is ready for review / merge!

**Review by ZLLentz — COMMENTED** [review]
This all seems reasonable to me!
I'm not in a position to evaluate the functionality but the code health looks good, I'm dropping a soft approval here.

File: scripts/startami2
Comment (ZLLentz) [inline scripts/startami2:1]: I think the main thing missing from my point of view is including this script in the readme table

**Comment by ZLLentz** [comment]
I have no objections 👍 but I suspect I'm not the right person to approve this one

**Review by silkenelson — APPROVED** [review]
Looks good to me.

**Pull request merged**